internal/internal_worker_base.go

```diff
@@ -301,7 +308,7 @@ func (bw *baseWorker) pollTask() {
 	var err error
 	var task interface{}
 
-	if bw.pollerAutoScaler != nil {
+	if bw.concurrencyAutoScaler != nil {
 		if pErr := bw.concurrency.PollerPermit.Acquire(bw.limiterContext); pErr == nil {
```

Comment:
nit: this looks like a leaking abstraction. This should be handled inside concurrencyAutoScaler.
I suggest moving all 
concurrencyAutoScaler != nil checks inside methods where it is required.
This code should be simpler. Just calling methods on autoscaler. If it is nil, do nothing.

Reply:
make sense

Reply:
we are guarding calls againts `bw.concurrency` based on nilness of `bw.concurrencyAutoScaler` which indicates that these two should be abstracted behind a single interface to avoid additional complexity in this file